### PR TITLE
refactor: removed async from dependency list

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes [#2883](https://github.com/googleapis/google-cloud-node/issues/2883) (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,7 +84,6 @@
     "@types/through2": "^2.0.34",
     "@types/time-span": "^2.0.0",
     "@types/uuid": "^3.4.4",
-    "async": "^2.6.1",
```
Comment (crwilcox) [inline package.json]: system-test.ts relies on this along with cleanup.js. We can move it dev dependencies though.

**Comment by JustinBeckwith** [comment]
aye, the work here is to strip the use of async out of those files :)

**Comment by vijay-qlogic** [comment]
@JustinBeckwith yes 'll update this PR, working on it, looking for how can I replace async.eachLimit (npm module) with native async/await

**Comment by JustinBeckwith** [comment]
I typically use p-limit or p-queue for that :)

**Comment by vijay-qlogic** [comment]
@JustinBeckwith Thanks! I used p-limit for async.eachLimit

File: package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "lodash.merge": "^4.6.1",
     "lodash.snakecase": "^4.1.1",
     "merge-stream": "^1.0.1",
+    "p-limit": "^2.0.0",
```
Comment (JustinBeckwith) [inline package.json]: You probably want to add `@types/p-limit` as well?

File: scripts/cleanup.js [hunk 3]
```diff
@@ -31,21 +30,15 @@ async function deleteStaleInstances(labelFilter) {
...
-          instance.delete(callback);
-        }, 500); // Delay allows the instance and its databases to fully clear.
-      },
-      (err, res) => {
-        if (err) return reject(err);
-        resolve(res);
-      }
-    );
+  const limit = pLimit(5);
+  const input = [];
```
Comment (JustinBeckwith) [inline scripts/cleanup.js]: I think this whole thing can be reduced to: 

```js
await Promise.all(
  filtered.map(instance => limit(instance.delete()))
);
```

File: system-test/spanner.ts [hunk 4]
```diff
@@ -45,15 +45,10 @@ describe('Spanner', () => {
...
-          deleteTestResources,
-          (next) => {
-            instance.create(INSTANCE_CONFIG, execAfterOperationComplete(next));
-          },
-        ],
-        done);
+  before(async () => {
+    await deleteTestResources;
+    await instance.create(INSTANCE_CONFIG);
+    await execAfterOperationComplete;
```
Comment (JustinBeckwith) [inline system-test/spanner.ts]: That doesn't look right :) What is `execAfterOperationComplete`?  Is that a promise?

File: system-test/spanner.ts [hunk 5]
```diff
@@ -873,25 +868,13 @@ describe('Spanner', () => {
...
-              session.create(next);
-            },
-          ],
-          done);
+    before(async () => {
+      await database.create({
+        schema: `CREATE TABLE Singers (SingerId STRING(1024) NOT NULL,
+          Name STRING(1024),) PRIMARY KEY(SingerId)`,
+      });
+      await execAfterOperationComplete;
```
Comment (JustinBeckwith) [inline system-test/spanner.ts]: Same as above, that don't look right

File: system-test/spanner.ts [hunk 6]
```diff
@@ -4246,29 +4221,23 @@ function execAfterOperationComplete(callback) {
...
-          setTimeout(() => {
-            // tslint:disable-next-line no-any
-            (instance as any).delete(callback);
-          }, 500);  // Delay allows the instance and its databases to fully
-                    // clear.
-        }, done);
-      });
+  const limit = pLimit(5);
+  const input: Array<{}> = [];
+  instances.forEach(instance => {
```
Comment (JustinBeckwith) [inline system-test/spanner.ts]: `await Promise.all(instances.map(limit(instance.delete())));`

File: system-test/spanner.ts [hunk 7]
```diff
@@ -4246,29 +4221,23 @@ function execAfterOperationComplete(callback) {
...
+    input.push(limit(async () => await instance.delete()));
+  });
+  return (async () => {
+    return await Promise.all(input);
+  })();
 }
 
-function deleteTestResources(callback) {
-  async.series([deleteTestInstances], callback);
+async function deleteTestResources() {
```
Comment (JustinBeckwith) [inline system-test/spanner.ts]: I don't see a reason for this method to exist if that's all it does :)

File: package.json (hunk 2 above)
Comment (vijay-qlogic) [reply package.json]: image
I face this error on adding "@type/p-limit" 
Not sure if I should add this...

File: scripts/cleanup.js (hunk 3 above)
Comment (vijay-qlogic) [reply scripts/cleanup.js]: Done!

File: system-test/spanner.ts (hunk 4 above)
Comment (vijay-qlogic) [reply system-test/spanner.ts]: It's not a Promise, should not call it this way. 
I'll check if I can promisify this and call then

File: package.json (hunk 2 above)
Comment (JustinBeckwith) [reply package.json]: Just means you need to import it using `import pLimit = require('p-limit');` which is admittedly confusing.

**Pull request merged**